### ISSUE TITLE
panics on startup with TryFromSliceError in byteorder crate

### ISSUE THREAD

**Issue by Gelbpunkt** [actix/actix-web#1890]
## Expected Behavior
The client should get a bad request or method not allowed.

## Current Behavior
The webserver panics.

```rust
thread 'actix-rt:worker:0' panicked at 'called `Result::unwrap()` on an `Err` value: TryFromSliceError(())', /home/jens/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.4.0/src/lib.rs:2185:43
```

## Steps to Reproduce (for bugs)

```rust
use actix_web::{middleware, post, web, App, HttpServer, Responder};

use std::env::{set_var, var};

#[post("/upload")]
async fn upload_package(mut payload: web::Payload) -> impl Responder {
    "hello world"
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    if var("RUST_LOG").is_err() {
        set_var("RUST_LOG", "info");
    }
    env_logger::init();

    HttpServer::new(|| {
        App::new()
            .wrap(middleware::Logger::default())
            .service(upload_package)
    })
    .bind("0.0.0.0:7575")?
    .run()
    .await
}
```
Then `curl localhost:7575/upload` to send a GET

## Context
Trying to make a package upload endpoint that only accepts POST and gives errors on GET.

## Your Environment

* Rust Version (I.e, output of `rustc -V`): rustc 1.51.0-nightly (44e3daf5e 2020-12-31)
* Actix Web Version: 4.0.0-beta.1 and git both same result

**Comment by robjtede** [comment]
I'm noticing that `byteorder-1.4.0` is new today. Investigating.

The line referencing in the panic message:
https://github.com/BurntSushi/byteorder/blob/1.4.0/src/lib.rs#L2185

**Comment by robjtede** [comment]
Workaround: add following to Cargo.toml:

```
byteorder = "~1.3"
```

**Comment by robjtede** [comment]
Fixed in byteorder v1.4.1. Run a cargo update to fix.
